### ISSUE TITLE
[Docs] Review migration rates documentation

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/msprime#988]
A persistent source of confusion in msprime has been what the migration in rates in msprime actually mean.

#978 updated the simulation model section with some clarifications from @petrelharp, and harmonised the usage of source/dest in the MigrationRateChange and MassMigration events. We should review this, and try to clarify as much as possible - perhaps adding a section in the tutorial to really emphasise the point with some simple examples, which we link to the in terse "simulation model" section.

**Comment by jeromekelleher** [comment]
Note also the discrepancy between migration rates in the DTWF and Hudson models, #1105

**Comment by nspope** [comment]
I am looking at [docs/api.rst](https://github.com/tskit-dev/msprime/blob/master/docs/api.rst#simulation-model) in relation to #1105 :

> Population structure is modelled by specifying a fixed number of subpopulations `d`, and a `d \times d` matrix `M` of per generation migration rates. Each element of the matrix `M_{j,k}` defines the fraction of population `j` that consists of migrants from population `k` in each generation. Note that migration rates are specified in a way that makes sense for the coalescence process, so that the `(j, k)^{th}` entry of the migration matrix, `M_{j, k}`, gives the rate at which an ancestral lineage moves from population `j` to population `k` as one follows it back through time.

The description of `M_{j,k}` as a "fraction" is confusing (to me) for continuous time. The way I think of it is that `M_{j,k}` is the expected number of migrations from `k` to `j`, per lineage per generation, forward in time. For discrete time `\sum_{k \neq j} M_{j,k}` has to be less than or equal to 1, because there is at most one migration event per lineage per generation. For continuous time, "generations" are just units and not a fundamental way to organize events: so migration per lineage can happen more "quickly" than 1/generation, and the rates aren't constrained to the unit simplex.

Does this seem like a reasonable change (from 'fraction' to 'expected migrations per individual per generation')? I am after a definition that applies equally well to DTWF and the coalescent.

**Comment by petrelharp** [comment]
Well, let's see, thinking through things: Suppose there's a population of size `N` and an inmigration rate of `mN`, i.e., at rate `m` new individuals show up from elsewhere and replace a randomly chosen individual. The expected number of replacements per unit time is `mN`, and since the rate at which migrants arrive and replace a *given* individual is `mN/N = m`, the probability a given individual is replaced is `1 - exp(-m) = m + O(m^2)`.

So, there's not going to be an exact correspondence between the 'migration rates' in the DTWF and Kingman's coalescent.

How about this:
```
Population structure is modelled by specifying a fixed number of subpopulations d, and a d \times d matrix M of migration rates,
in units of migrations per generation. In terms of the coalescent process,
the (j, k)^{th} entry of the migration matrix, M_{j, k}, gives the rate at which an ancestral lineage moves from population j to population k as one follows it back through time.
This is equal to the mean number ofmigrants from population k to population j per generation, divided by the size of population j.
In discrete time models, this is also equal to the fraction of population j that consists migrants from population k per generation,
and is approximately equal to this in continuous-time models.
```

Gee, that's a bit awkward still. Want to edit it?

**Comment by nspope** [comment]
I think that's an improvement! But the distinction I'm getting at is that rates can exceed 1 in the continuous time models -- and of course when rates are greater than say 0.2-0.3, the higher order terms in that series expansion aren't tiny. So perhaps:

>Population structure is modelled by specifying a fixed number of subpopulations d, and a d \times d matrix M of per-generation migration rates. The (j,k)^th entry of M is the expected number of migrants moving from population k to population j per generation, divided by the size of population j. In terms of the coalescent process, M_{j,k} gives the rate at which an ancestral lineage moves from population j to population k, moving backwards through time. In continuous-time models, when M_{j,k} << 1, it is approximately equal to the fraction of population j that is replaced each generation by migrants from population k. In discrete-time models, this equivalence is exact and each row of M has the constraint \sum_{k \neq j} M_{j,k} \leq 1.

**Comment by petrelharp** [comment]
This sounds good to me, as long as the "when M << 1" caveat is not too confusing to people who haven't thought about this math.  And, I strongly vote to say "as one follows it back through time" instead of "moving backwards through time".

**Comment by jeromekelleher** [comment]
Can we close this issue now do you think @petrelharp? Or do we still need to work on these docs?

**Comment by petrelharp** [comment]
Yes, the docs fixed up in #1105 look great!
